python/services/evaluationservice/dmod/evaluationservice/utilities/common.py

```diff
@@ -15,6 +17,48 @@ def key_separator() -> str:
     return "--"
 
 
+def create_basic_credentials(
+    username: typing.Union[str, bytes],
+    password: typing.Union[str, bytes]
+) -> typing.Mapping[typing.Literal["HTTP_AUTHORIZATION"], str]:
+    """
+    Creates a key-value pair with the appropriate header key and header value for basic credential authentication
+
+    Args:
+        username: The username for the credentials
+        password: The password for the credentials
+
+    Returns:
+        A mapping from the header key to the header value
+    """
+    if isinstance(username, bytes):
+        username = username.decode()
+
+    if isinstance(password, bytes):
+        password = password.decode()
+
+    username_and_password = f"{username}:{password}".encode()
+    return {"HTTP_AUTHORIZATION": f"Basic {base64.b64encode(username_and_password).decode()}"}
```

Comment:
It might be worth adding a comment here noting that this is a django specific keyword argument that gets turned into an `Authorization` header.